OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs

```diff
@@ -45,13 +45,15 @@ class HarvesterTraitWrapper
 			public readonly Harvester Harvester;
 			public readonly Parachutable Parachutable;
 			public readonly LocomotorInfo LocomotorInfo;
+			public readonly Locomotor Locomotor;
 
 			public HarvesterTraitWrapper(Actor actor)
 			{
 				Actor = actor;
 				Harvester = actor.Trait<Harvester>();
 				Parachutable = actor.TraitOrDefault<Parachutable>();
 				LocomotorInfo = actor.Info.TraitInfo<MobileInfo>().LocomotorInfo;
+				Locomotor = actor.TraitOrDefault<Mobile>().Locomotor;
```

Comment:
This pattern does appear to repeat itself quite a bit. Are you sure you can't access `MobileInfo` from the `Mobile` trait?
If you can, then please replace all `TraitInfo<MobileInfo>()` queries by `mobile.Info` to save a query.
While we are at it: Don't use `TraitOrDefault` if the expected result shouldn't be `null` (which in this case would NRE). Use just `Trait<Mobile>` instead.